Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGrid.cs

```diff
@@ -1263,6 +1266,92 @@ private static void OnIsValidPropertyChanged(DependencyObject d, DependencyPrope
             }
         }
 
+        // Field used for accessing the ItemsSource though the interface ISupportIncrementalLoading
+        private ISupportIncrementalLoading _incrementalItemsSource;
+
+        private bool _loadingItems;
+
+        // Method that loads additional data once the user has scrolled to a certain threashold
+        private async void LoadMoreDataFromIncrementalItemsSource()
+        {
+            if (IncrementalLoadingTrigger == IncrementalLoadingTrigger.Edge && _incrementalItemsSource != null && _incrementalItemsSource.HasMoreItems && !_loadingItems)
+            {
+                var bottomScrollOffHeight = Math.Max(0, EdgedRowsHeightCalculated - CellsHeight - VerticalOffset);
+
+                var rowsPresenterAvailableHeight = this._rowsPresenterAvailableSize.HasValue
+                    ? _rowsPresenterAvailableSize.Value.Height
+                    : _vScrollBar.ViewportSize;
+
+                if ((IncrementalLoadingThreshold * rowsPresenterAvailableHeight) >= bottomScrollOffHeight)
+                {
+                    _loadingItems = true;
+
+                    var numberOfRowsToLoad = (uint)(DataFetchSize * rowsPresenterAvailableHeight / this.RowHeightEstimate);
+
+                    await _incrementalItemsSource.LoadMoreItemsAsync(numberOfRowsToLoad);
+
+                    _loadingItems = false;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Identifies the IncrementalLoadingThreshold dependency property
+        /// </summary>
+        public static readonly DependencyProperty IncrementalLoadingThresholdProperty =
+            DependencyProperty.Register(
+                "IncrementalLoadingThreshold",
+                typeof(double),
+                typeof(DataGrid),
+                new PropertyMetadata(3.0));
+
+        /// <summary>
+        /// Gets or sets the IncrementalLoadingThreshold
+        /// </summary>
+        public double IncrementalLoadingThreshold
+        {
+            get { return (double)GetValue(IncrementalLoadingThresholdProperty); }
+            set { SetValue(IncrementalLoadingThresholdProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the IncrementalLoadingTrigger dependency property
+        /// </summary>
+        public static readonly DependencyProperty IncrementalLoadingTriggerProperty =
+            DependencyProperty.Register(
+                "IncrementalLoadingTrigger",
+                typeof(IncrementalLoadingTrigger),
+                typeof(DataGrid),
+                new PropertyMetadata(IncrementalLoadingTrigger.Edge));
+
+        /// <summary>
+        /// Gets or sets the IncrementalLoadingTrigger
```

Comment:
Let's match the ListViewBase comments
        //
        // Summary:
        //     Gets or sets a value that indicates the conditions for prefetch operations by
        //     the GridView class.
        //
        // Returns:
        //     An enumeration value that indicates the conditions that trigger prefetch operations.
        //     The default is **Edge**.